src/middlewared/middlewared/plugins/activedirectory.py

```diff
@@ -619,57 +678,105 @@ class Config:
+        for each trusted domain, and the idmap cache should be cleared.
+
+        `allow_dns_updates` during the domain join process, automatically
+        generate DNS entries in the AD domain for the NAS. If this is disabled,
+        then a domain administrator must manually add appropriate DNS entries
+        for the NAS. This is a recommended parameter for TrueNAS HA servers.
+
+        `disable_freenas_cache` disable active caching of AD users and groups.
+        When disabled, only users cached in winbind's internal cache will be
+        visible in GUI dropdowns. Disabling active caching is recommended
+        environments with a large amount of users.
+
+        `site` AD site of which the NAS is a member. This parameter is auto-
+        detected during the domain join process. If no AD site is configured
+        for the subnet in which the NAS is configured, then this parameter will
+        appear as 'Default-First-Site-Name'. Auto-detection is only performed
+        during the initial domain join.
+
+        `kerberos_realm` in which the server is located. This parameter will be
+        automatically populated during the initial domain join. If the NAS has
+        an AD site configured and that site has multiple kerberos servers, then
+        the kerberos realm will be automatically updated with a site-specific
+        configuration to use those servers. Auto-detection is only performed
+        during initial domain join.
+
+        `kerberos_principal` kerberos principal to use for AD-related
+        operations outside of Samba. After intial domain join, this field will
+        be updated with the kerberos principal associated with the AD machine
+        account for the NAS.
+
+        `nss_info` controls how Winbind retrieves Name Service Information to
+        construct a user's home directory and login shell. This parameter
+        is only effective if the Active Directory Domain Controller supports
+        the Microsoft Services for Unix (SFU) LDAP schema.
+
+        `timeout` timeout value for winbind-related operations. This value may
+        need to be increased in  environments with high latencies for
+        communications with domain controllers or a large number of domain
+        controllers. Lowering the value may cause status checks to fail.
+
+        `dns_timeout` timeout value for DNS queries during the initial domain
+        join. This value is also set as the NETWORK_TIMEOUT in the ldap config
+        file.
+
+        `ldap_sasl_wrapping` defines whether ldap traffic will be signed or
+        signed and encrypted (sealed). LDAP traffic that does not originate
+        from Samba defaults to using GSSAPI signing unless it is tunnelled
+        over LDAPs.
+
+        `createcomputer` Active Directory Organizational Unit in which to
+        create the NAS computer object during domain join.
+
+        If blank, then the default OU is used during computer account creation.
```

Comment:
```
        If blank, then the default OU is used during computer account creation.
```

Same as the beginning. I would expand this first mention of the `OU` acronym. I'm not sure what it is.

Reply:
I think Warren wrote up something that put this in understandable language. So this entry should probably be rewritten using the authorized text.